types/time.go

```diff
@@ -1214,11 +1214,26 @@ func ParseDuration(sc *stmtctx.StatementContext, str string, fsp int8) (Duration
 		sign = -1
 	}
 
-	// Time format may has day.
+	// Time format may have leading and trailing whitespaces.
+	str = strings.TrimSpace(str)
+
+	// Time format may have day.
 	if n := strings.IndexByte(str, ' '); n >= 0 {
-		if day, err = strconv.Atoi(str[:n]); err == nil {
-			dayExists = true
+		day, err = strconv.Atoi(str[:n])
+		// Time format may have date
+		if err != nil {
+			t, err1 := ParseDatetime(sc, str)
```

Comment:
Could you please explain why call `ParseDatetime` here 👀?

Reply:
Please have a look at this examples:
"1ad 10:10:10" - this one is not correct
"2019-01-12 10:10:10" - this one is correct
```sh
mysql> select time("1ad 10:13:12");
+----------------------+
| time("1ad 10:13:12") |
+----------------------+
| NULL                 |
+----------------------+
1 row in set, 1 warning (0.01 sec)

mysql> select time("2012-12-11 10:13:12");
+-----------------------------+
| time("2012-12-11 10:13:12") |
+-----------------------------+
| 10:13:12                    |
+-----------------------------+
1 row in set (0.00 sec)
```
but in both cases strconv.Atoi returns an error. Previously the error was dismissed and 1st example was considered as correct (just without a day)

Reply:
got.